akka-stream/src/main/scala/akka/stream/impl/io/SslTls.scala

```diff
@@ -69,8 +69,8 @@ private[akka] class SslTlsCipherActor(settings: ActorFlowMaterializerSettings, s
    * block.
    */
   class ChoppingBlock(idx: Int, name: String) extends TransferState {
-    override def isReady: Boolean = buffer.nonEmpty
-    override def isCompleted: Boolean = false
+    override def isReady: Boolean = buffer.nonEmpty || inputBunch.isPending(idx) || inputBunch.isDepleted(idx)
+    override def isCompleted: Boolean = inputBunch.isCancelled(idx)
```

Comment:
Yes, and I’m saying this on the record: you were exactly right :-) The part that this handles gracefully is when termination happens “unexpectedly”, as is the case for `fail(e)`.